pkg/process/checks/net.go

```diff
@@ -126,12 +131,82 @@ func (c *ConnectionsCheck) Realtime() bool { return false }
 // ShouldSaveLastRun indicates if the output from the last run should be saved for use in flares
 func (c *ConnectionsCheck) ShouldSaveLastRun() bool { return false }
 
+func (c *ConnectionsCheck) handleBatch(batch *model.Connections, start time.Time, groupID int32, isFirst bool) model.MessageBody {
+	// Resolve the Raddr side of connections for local containers
+	LocalResolver.Resolve(batch)
+
+	c.notifyProcessConnRates(c.config, batch)
+
+	log.Debugf("collected connections in %s", time.Since(start))
+
+	return processConnectionsBatch(c.hostInfo, c.maxConnsPerMessage, groupID, batch.Conns, batch.Dns, c.networkID,
+		batch.ConnTelemetryMap, batch.CompilationTelemetryByAsset, batch.KernelHeaderFetchResult,
+		batch.CORETelemetryByAsset, batch.PrebuiltEBPFAssets, batch.Domains, batch.Routes, batch.Tags,
+		batch.AgentConfiguration, c.serviceExtractor, isFirst)
+
+}
+
 // Run runs the ConnectionsCheck to collect the active network connections
 // and any closed network connections since the last Run.
 // For each connection we'll return a `model.Connection`
 // that will be bundled up into a `CollectorConnections`.
 // See agent.proto for the schema of the message and models.
 func (c *ConnectionsCheck) Run(nextGroupID func() int32, _ *RunOptions) (RunResult, error) {
+	if c.syscfg.GRPCServerEnabled {
+		// Create a context with a timeout of 10 seconds
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+
+		conn, err := grpc.Dial("unix://"+c.syscfg.GRPCSocketFilePath, grpc.WithInsecure())
```

Comment:
it think we should match the behavior to http so windows flavor will be able to use the feature